subxt/src/client/lightclient/rpc.rs

```diff
@@ -0,0 +1,215 @@
+    to_backend: mpsc::Sender<FromSubxt>,
+}
+
+impl LightClientRpc {
+    /// Constructs a new [`LightClientRpc`], providing the chain specification.
+    ///
+    /// The chain specification can be downloaded from a trusted network via
+    /// the `sync_state_genSyncSpec` RPC method. This parameter expects the
+    /// chain spec in text format (ie not in hex-encoded scale-encoded as RPC methods
+    /// will provide).
+    ///
+    /// ## Panics
+    ///
+    /// Panics if being called outside of `tokio` runtime context.
+    pub fn new(
+        config: smoldot_light::AddChainConfig<'_, (), impl Iterator<Item = ChainId>>,
+    ) -> Result<LightClientRpc, Error> {
+        tracing::trace!(target: LOG_TARGET, "Create light client");
+
+        let mut client = smoldot_light::Client::new(Platform::new(
+            env!("CARGO_PKG_NAME").into(),
+            env!("CARGO_PKG_VERSION").into(),
+        ));
+
+        let smoldot_light::AddChainSuccess {
+            chain_id,
+            json_rpc_responses,
+        } = client
+            .add_chain(config)
+            .map_err(|err| LightClientError::AddChainError(err.to_string()))?;
+
+        let (to_backend, backend) = mpsc::channel(128);
+
+        // `json_rpc_responses` can only be `None` if we had passed `json_rpc: Disabled`.
+        let rpc_responses = json_rpc_responses.expect("Light client RPC configured; qed");
+
+        tokio::spawn(async move {
+            let mut task = BackgroundTask::new(client, chain_id);
+            task.start_task(backend, rpc_responses).await;
+        });
+
+        Ok(LightClientRpc { to_backend })
+    }
+
+    /// Submits an RPC method request to the light-client.
+    ///
+    /// This method sends a request to the light-client to execute an RPC method with the provided parameters.
+    /// The parameters are parsed into a valid JSON object in the background.
+    async fn method_request(
+        &self,
+        method: String,
+        params: String,
+    ) -> Result<oneshot::Receiver<MethodResponse>, SendError<FromSubxt>> {
+        let (sender, receiver) = oneshot::channel();
+
+        self.to_backend
+            .send(FromSubxt::Request {
+                method,
+                params,
+                sender,
+            })
+            .await?;
+
+        Ok(receiver)
+    }
+
+    /// Makes an RPC subscription call to the light-client.
+    ///
+    /// This method sends a request to the light-client to establish an RPC subscription with the provided parameters.
+    /// The parameters are parsed into a valid JSON object in the background.
+    async fn subscription_request(
+        &self,
+        method: String,
+        params: String,
+    ) -> Result<
+        (
+            oneshot::Receiver<MethodResponse>,
+            mpsc::Receiver<Box<RawValue>>,
+        ),
+        SendError<FromSubxt>,
+    > {
+        let (sub_id, sub_id_rx) = oneshot::channel();
+        let (sender, receiver) = mpsc::channel(128);
```

Comment:
could be good make this configurable at some point, just a warning that backpressure deadlocks are super hard to detect/debug.

Initially I would make these unbounded but your call :)